registration_api.js

```diff
@@ -0,0 +1,596 @@
+    : new InScriptStore();
+
+const CookieStorage = function (cookie_name) {
+    this.initialized = false;
+    this.cookie_name = cookie_name;
+    this.domain = 'api.deriv.com';
+    this.path = '/';
+    this.expires = new Date('Thu, 1 Jan 2037 12:00:00 GMT');
+    this.value = {};
+};
+
+CookieStorage.prototype = {
+    read() {
+        const cookie_value = Cookies.get(this.cookie_name);
+        try {
+            this.value = cookie_value ? JSON.parse(cookie_value) : {};
+        } catch (e) {
+            this.value = {};
+        }
+        this.initialized = true;
+    },
+    write(val, expireDate, isSecure) {
+        if (!this.initialized) this.read();
+        this.value = val;
+        if (expireDate) this.expires = expireDate;
+        Cookies.set(this.cookie_name, this.value, {
+            expires: this.expires,
+            path: this.path,
+            domain: this.domain,
+            secure: !!isSecure,
+        });
+    },
+    get(key) {
+        if (!this.initialized) this.read();
+        return this.value[key];
+    },
+    set(key, val) {
+        if (!this.initialized) this.read();
+        this.value[key] = val;
+        Cookies.set(this.cookie_name, this.value, {
+            expires: new Date(this.expires),
+            path: this.path,
+            domain: this.domain,
+        });
+    },
+    remove() {
+        Cookies.remove(this.cookie_name, {
+            path: this.path,
+            domain: this.domain,
+        });
+    },
+};
+
+const appId = () => window.localStorage.getItem("config.app_id");
+
+
+const loginUrl = ({ language }) => {
+    const server_url = LocalStore.get('config.server_url');
+    const signup_device_cookie = new CookieStorage('signup_device');
+    const signup_device = signup_device_cookie.get('signup_device');
+    const date_first_contact_cookie = new CookieStorage('date_first_contact');
+    const date_first_contact = date_first_contact_cookie.get('date_first_contact');
+    const marketing_queries = `${signup_device ? `&signup_device=${signup_device}` : ''}${date_first_contact ? `&date_first_contact=${date_first_contact}` : ''
+        }`;
+    const getOAuthUrl = () => {
+        return `https://oauth.deriv.com/oauth2/authorize?app_id=${appId()}&l=${language}${marketing_queries}&brand=deriv`;
+    };
+
+    if (server_url && /qa/.test(server_url)) {
+        return `https://${server_url}/oauth2/authorize?app_id=${appId()}&l=${language}${marketing_queries}&brand=deriv`;
+    }
+
+    if (window.localStorage.getItem("config.app_id") === 11780) {
+        return getOAuthUrl();
+    }
+    return new URL(getOAuthUrl()).href;
+};
+
+let sessionState = sessionStorage.getItem('app_registration_state') || 'logged_out';
+const urlParams = new URLSearchParams(window.location.search);
+const token = urlParams.get('token1');
+if (token) {
+    sessionStorage.setItem('token1', token);
+    sessionStorage.setItem('app_registration_state', 'logged_in');
+    sessionState = 'logged_in';
+}
+
```

Comment:
This functional suppose to be outside any function?

Reply:
Yes, as it can be treated as onMount / useEffect on load type of requirement (when this view loads, if the token1 exists in the url => it means that login was successful and we are saving the token to sessionStorate. It executes only once)